--- coffea/processor/executor_.py	2021-02-04 18:27:37.000000000 +0100
+++ coffea/processor/executor.py	2021-02-04 18:27:43.000000000 +0100
@@ -42,6 +42,7 @@
 _PICKLE_PROTOCOL = pickle.HIGHEST_PROTOCOL
 DEFAULT_METADATA_CACHE = LRUCache(100000)
 
+uproot4.open.defaults["xrootd_handler"] = uproot4.source.xrootd.MultithreadedXRootDSource
 
 class FileMeta(object):
     __slots__ = ['dataset', 'filename', 'treename', 'metadata']
