diff --git a/bindings/python/CMakeLists.txt b/bindings/python/CMakeLists.txt
index 0aa53ee5e..6d2d11b6b 100644
--- a/bindings/python/CMakeLists.txt
+++ b/bindings/python/CMakeLists.txt
@@ -38,7 +38,7 @@ STRING(REGEX REPLACE "\".*"     "" DEB_DISTRO "${DEB_DISTRO}")
 
 if( DEB_DISTRO STREQUAL "Debian" OR DEB_DISTRO STREQUAL "Ubuntu" )
   set(PYTHON_LAYOUT  "unix" CACHE STRING "Python installation layout (deb or unix)")
-  set(DEB_INSTALL_ARGS "--install-layout ${PYTHON_LAYOUT}")
+  #set(DEB_INSTALL_ARGS "--install-layout ${PYTHON_LAYOUT}")
 endif()
 
 install(
