--- reading.py	2021-07-09 14:09:34.000000000 +0200
+++ reading_patched.py	2021-07-09 14:09:31.000000000 +0200
@@ -164,7 +164,7 @@
 
 open.defaults = {
     "file_handler": uproot.source.file.MemmapSource,
-    "xrootd_handler": uproot.source.xrootd.XRootDSource,
+    "xrootd_handler": uproot.source.xrootd.MultithreadedXRootDSource,
     "http_handler": uproot.source.http.HTTPSource,
     "object_handler": uproot.source.object.ObjectSource,
     "timeout": 30,
