diff --git a/src/XrdCl/CMakeLists.txt b/src/XrdCl/CMakeLists.txt
index 759e7c171..9d1550b32 100644
--- a/src/XrdCl/CMakeLists.txt
+++ b/src/XrdCl/CMakeLists.txt
@@ -171,6 +171,7 @@ install(
     XrdClStatus.hh
     XrdClURL.hh
     XrdClXRootDResponses.hh
+    XrdClOptional.hh
     XrdClPlugInInterface.hh
     XrdClPlugInManager.hh
     XrdClPropertyList.hh
diff --git a/utils/xrootd-config b/utils/xrootd-config
index 3cce14c09..685fdc8d3 100755
--- a/utils/xrootd-config
+++ b/utils/xrootd-config
@@ -39,7 +39,7 @@ Generic options
 
 Compilation support options
   --cflags      print pre-processor and compiler flags
-  --plugin-vesion print plug-in version suffix
+  --plugin-version print plug-in version suffix
 
 Install directories XRootD was configured to
   --prefix
